lib/thrift/exceptions.ex

```diff
@@ -22,3 +22,29 @@ defmodule Thrift.Union.TooManyFieldsSetException do
   """
   defexception message: nil, set_fields: nil
 end
+
+defmodule Thrift.FileParseException do
+  @moduledoc """
+  This exception occurs when a thrift file fails to parse
+  """
+
+  defexception message: nil
+
+  @doc false  # Exception callback, should not be called by end user
+  @spec exception({Thrift.Parser.FileRef.t, term}) :: Exception.t
+  def exception({file_ref, error}) do
+    msg = "Error parsing thrift file #{file_ref.path} #{format_error(error)}"
+    %Thrift.FileParseException{message: msg}
+  end
+
+  # display the line number if we get it
+  defp format_error({line_no, :thrift_parser, errors}) do
+    "on line #{line_no}: #{inspect errors}"
+  end
+  defp format_error({{line_no, :thrift_lexer, errors}, _}) do
+    "on line #{line_no}: #{inspect errors}"
+  end
+  defp format_error(error) do
+    ":#{inspect error}"
```

Comment:
This message won't be formatted quite right. There will be a space before the `:` instead of after.

The errors look suspiciously like iolists. I wonder if that is consistently true. Would be a bit nicer to display `syntax error before: "stract"` than `['syntax error before: ', ['"stract"']]`.